File: js/models/ProtocolType.js
(function(exports) {
    const ProtocolType = {
        ICMP: { code: 'ICMP', name: 'ICMP Ping / Echo', color: '#3b82f6', description: 'Network Control & Echo Diagnostics' },
        TCP_SYN: { code: 'TCP-SYN', name: 'TCP SYN', color: '#10b981', description: 'TCP Handshake Initiation' },
        TCP_ACK: { code: 'TCP-ACK', name: 'TCP ACK', color: '#34d399', description: 'TCP Acknowledgment' },
        HTTP: { code: 'HTTP', name: 'HTTP Web Request/Reply', color: '#f59e0b', description: 'Hypertext Transfer Protocol (Port 80)' },
        HTTPS: { code: 'HTTPS', name: 'HTTPS Secure Web', color: '#d97706', description: 'Encrypted Web Traffic (Port 443)' },
        DNS: { code: 'DNS', name: 'DNS Query/Response', color: '#a855f7', description: 'Domain Name System (Port 53)' },
        DHCP: { code: 'DHCP', name: 'DHCP IP Lease', color: '#ec4899', description: 'Dynamic Host Configuration Protocol' },
        FTP: { code: 'FTP', name: 'FTP File Transfer', color: '#0ea5e9', description: 'File Transfer Protocol (Port 21)' },
        ARP: { code: 'ARP', name: 'ARP Resolution', color: '#eab308', description: 'Address Resolution Protocol Broadcast' },
        DROPPED: { code: 'DROP', name: 'Dropped / Filtered', color: '#ef4444', description: 'Packet Dropped or Blocked by Firewall' }
    };

    exports.ProtocolType = ProtocolType;
})(typeof module !== 'undefined' && module.exports ? module.exports : (window.CampusNet = window.CampusNet || {}));

File: src/com/campusnet/model/ProtocolType.java
package com.campusnet.model;

import java.awt.Color;

public enum ProtocolType {
    ICMP("ICMP", "Ping / Echo Request & Reply", new Color(59, 130, 246)),
    TCP_SYN("TCP-SYN", "TCP Connection Initiation", new Color(16, 185, 129)),
    TCP_ACK("TCP-ACK", "TCP Acknowledgment", new Color(52, 211, 153)),
    HTTP("HTTP", "Web Request / Response (Port 80)", new Color(245, 158, 11)),
    HTTPS("HTTPS", "Secure Web (Port 443)", new Color(217, 119, 6)),
    DNS("DNS", "Domain Name Query / Response (Port 53)", new Color(168, 85, 247)),
    DHCP("DHCP", "IP Address Configuration", new Color(236, 72, 153)),
    FTP("FTP", "File Transfer Protocol (Port 21)", new Color(14, 165, 233)),
    ARP("ARP", "Address Resolution Protocol (Broadcast)", new Color(234, 179, 8)),
    DROPPED("DROP", "Packet Dropped / Blocked by Firewall", new Color(239, 68, 68));

    private final String code;
    private final String description;
    private final Color color;

    ProtocolType(String code, String description, Color color) {
        this.code = code;
        this.description = description;
        this.color = color;
    }

    public String getCode() {
        return code;
    }

    public String getDescription() {
        return description;
    }

    public Color getColor() {
        return color;
    }
}
